crypto/engine/eng_init.c

```diff
@@ -53,9 +54,8 @@ int engine_unlocked_finish(ENGINE *e, int unlock_for_handlers)
      * there's a chance that both threads will together take the count from 2
      * to 0 without either calling finish().
      */
-    e->funct_ref--;
-    engine_ref_debug(e, 1, -1);
-    if ((e->funct_ref == 0) && e->finish) {
+    ENGINE_DOWN_FUNCT_REF(e, &ref_cnt);
```

Comment:
No!  Same reason

Reply:
I think this one is safe.  The atomic decrement will keep the count consistent -- the comment is wrong and I'll remove.

I seem to remember that this is only called with the global lock held.

Reply:
Sorry about this one, I was going to remove this comment for exactly that reason, but forgot